### PR TITLE
use func keyword in system

### PR DESCRIPTION
**Pull request by cooldome**
Proc signatures are now shorter and easier to read, minus 130 tokens for parser to read.

### CODE REVIEW

**Comment by krux02** [comment]
Well easier to read is subjective. As a compiler developer I am far more used to the `proc` signature. But you are by far not the first one who wants `func` to be used in the compiler. I think I could get used to it as well, even though I am not really a big fan of that keyword.

**Review by dom96 — APPROVED** [review]
This is just a case of something that we need to get used to IMO. I think it makes sense to start introducing this.

File: compiler/semcall.nim [hunk 1]
```diff
@@ -657,7 +657,8 @@ proc searchForBorrowProc(c: PContext, startScope: PScope, fn: PSym): PSym =
       x = t.baseOfDistinct
     call.add(newNodeIT(nkEmpty, fn.info, x))
   if hasDistinct:
-    var resolved = semOverloadedCall(c, call, call, {fn.kind}, {})
+    let kind = if fn.kind == skProc: {skProc, skFunc} else: {fn.kind}
```
Comment (Araq) [inline compiler/semcall.nim]: I don't like this change, forward decls should be consistent.

Comment (cooldome) [reply compiler/semcall.nim]: ok

File: lib/pure/typetraits.nim [hunk 2]
```diff
@@ -64,8 +64,17 @@ proc supportsCopyMem*(t: typedesc): bool {.magic: "TypeTrait".}
   ##
   ## Other languages name a type like these `blob`:idx:.
 
-proc isNamedTuple*(T: typedesc): bool {.magic: "TypeTrait".}
-  ## Return true for named tuples, false for any other type.
+proc isNamedTuple(T: typedesc): bool =
```
Comment (Clyybber) [inline lib/pure/typetraits.nim]:
```suggestion
proc isNamedTuple*(T: typedesc): bool =
```
This should be exported right?

Comment (cooldome) [reply lib/pure/typetraits.nim]: Oops, thanks for checking

Comment (Clyybber) [reply lib/pure/typetraits.nim]: np, I think it may be the cause of the CI failures..

File: lib/system/dollars.nim [hunk 3]
```diff
@@ -144,11 +144,11 @@ func `$`*[T, U](x: HSlice[T, U]): string =
 
 
 when not defined(nimNoArrayToString):
-  func `$`*[T, IDX](x: array[IDX, T]): string =
+  proc `$`*[T, IDX](x: array[IDX, T]): string =
     ## Generic ``$`` operator for arrays that is lifted from the components.
     collectionToString(x, "[", ", ", "]")
 
-func `$`*[T](x: openArray[T]): string =
+proc `$`*[T](x: openArray[T]): string =
```
Comment (Clyybber) [inline lib/system/dollars.nim]: Why can't those be funcs?

**Comment by cooldome** [comment]
I have question. Looks like nimsuggest tests are posix only.
Is there a way to run nimsuggest tests on windows?

**Comment by Araq** [comment]
> Is there a way to run nimsuggest tests on windows?

There is no known way except to run some tests on the command line manually and to check the output.

**Pull request closed without merging**